pkg/providers/vsphere/vmprovider_vm.go

```diff
@@ -1102,6 +1111,47 @@ func (vs *vSphereVMProvider) reconcilePowerState(
 	return nil
 }
 
+func (vs *vSphereVMProvider) reconcileSnapshot(
+	vmCtx pkgctx.VirtualMachineContext,
+	vcVM *object.VirtualMachine) error {
+	// no need for snapshot or current snapshot is up-to-date
```

Comment:
Something along the lines of..
```suggestion
	// Skip creating a new snapshot if the VM already points to the desired snapshot.
```
?